### PR TITLE
Docker image enhancements

### PR DESCRIPTION
**Pull request by aslo**
This PR modifies the builder's docker images to use gcr.io/cloud-builders/java/mvn as the base image, which has a lighter footprint than the gcr.io/google-appengine/openjdk runtime, and contains pre-cached maven dependencies.

These changes result in a reduction in the size of the runtime-builder image from 448MB to 341MB, despite also including pre-cached maven deps.

### CODE REVIEW

File: java-runtime-builder-docker/src/main/resources/docker/Dockerfile [hunk 1]
```diff
@@ -1,29 +1,21 @@
...
-
-# install gradle
+# Use wget container to download gradle
+FROM gcr.io/cloud-builders/wget as wget
 ARG GRADLE_VERSION=3.4
 RUN mkdir -p /usr/share/gradle \
-  && curl -fsSLO https://services.gradle.org/distributions/gradle-$GRADLE_VERSION-bin.zip \
-  && unzip -qq gradle-$GRADLE_VERSION-bin.zip \
-  && rm gradle-$GRADLE_VERSION-bin.zip \
+  && wget -nv -O gradle-$GRADLE_VERSION-bin.zip https://services.gradle.org/distributions/gradle-$GRADLE_VERSION-bin.zip
```
Comment (meltsufin) [inline java-runtime-builder-docker/src/main/resources/docker/Dockerfile]: Why not just use `ADD https://services.gradle.org/distributions/gradle-$GRADLE_VERSION-bin.zip gradle-$GRADLE_VERSION-bin.zip`? The `FROM` line can then just be `scratch`?

File: java-runtime-builder-docker/src/main/resources/docker/Dockerfile [hunk 2]
```diff
@@ -1,29 +1,21 @@
...
+
+FROM gcr.io/cloud-builders/java/mvn:3.5.0-jdk-8
+ARG GRADLE_VERSION=3.4
+# copy gradle installation from wget container
+COPY --from=wget gradle-$GRADLE_VERSION-bin.zip .
+# unpack and install gradle
+RUN unzip -qq gradle-$GRADLE_VERSION-bin.zip \
```
Comment (meltsufin) [inline java-runtime-builder-docker/src/main/resources/docker/Dockerfile]: I think ADD can automatically unzip.

Comment (aslo) [inline java-runtime-builder-docker/src/main/resources/docker/Dockerfile]: I think you're right, but I believe only the COPY command supports the `--from` argument to copy from another build step. I'm pretty sure we can't avoid explicitly unzipping.

File: java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/MavenBuildStep.java [hunk 3]
```diff
@@ -53,7 +53,7 @@ protected void doBuild(Path directory, Map<String, String> metadata) throws Buil
 
   @VisibleForTesting
   String getMavenHome() {
-    return System.getenv("M2_HOME");
+    return System.getenv("MAVEN_HOME");
```
Comment (meltsufin) [inline java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/MavenBuildStep.java]: Why the change? I believe `M2_HOME` is the convention since Maven 2. Not important though.

Comment (aslo) [inline java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/MavenBuildStep.java]: The new base image sets `MAVEN_HOME` and not `M2_HOME`, so this needed to change as well. I agree, probably not the most important thing.
